src/app/modules/products/products-list/products-list.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { HttpClientModule } from '@angular/common/http';
+import { take } from 'rxjs';
+import { MatPaginatorModule } from '@angular/material/paginator'
+
+import { ProductsService } from '../products.service';
+import { ProductQuickviewComponent } from '../product-quickview/product-quickview.component';
+import { IProduct } from '../iproduct';
+
+@Component({
+  selector: 'app-products-list',
+  standalone: true,
+  imports: [
+    CommonModule,
+    MatPaginatorModule,
+    ProductQuickviewComponent,
+    HttpClientModule
+  ],
+  templateUrl: './products-list.component.html',
+  styleUrls: ['./products-list.component.scss']
+})
+export class ProductsListComponent implements OnInit {
+
+  constructor(public productsService: ProductsService) {
+  }
+
+  ngOnInit(): void {
+
+    this.productsService.products$
+      .subscribe({
+        next: (products: IProduct[]) => {
```

Comment:
we can refactor this to a private method and call it with this.myPrivateMethod.bind(this)

this pattern perserves functionality, divides code logically and facilitates testing each step independently